app.module.js

```diff
@@ -0,0 +1,163 @@
+    var thought = t[Math.floor(Math.random() * t.length)];
+    vm.thought = {
+      post: thought.post,
+      author: 'u/' + thought.author,
+      link: 'http://reddit.com' + thought.permalink
+    }
+    vm.view.innerHTML = vm.renderView('main_tmpl', vm.thought)
+  }
+
+  if ((cachedThoughts &&
+    new Date() <= new Date(vm.state.cachedPosts.expirationCacheTime)) ||
+    !navigator.onLine
+  ) {
+    getAndRender(vm.state.cachedPosts.data)
+  } else {
+    this.fetchData('https://www.reddit.com/r/showerthoughts/hot.json?limit=300')
+    .then(function (res) {
+      Object.assign(vm.state.cachedPosts, {
+        data: res,
+        expirationCacheTime: new Date().addHours(1)
+      });
+      getAndRender(vm.state.cachedPosts.data);
+      vm.saveState(vm.state);
+    })
+    .catch(function (err) {
+      throw new Error(err);
+    })
+  }
+}
+
+App.prototype.fetchData = function (url) {
+  return new Promise(function (resolve, reject) {
+    if (url) {
+      fetch(url)
+      .then(function(data) {
+        return data.json()
+      })
+      .then(function(res) {
+        var data = res.data.children.map(function(post) {
+          return {
+            post: post.data.title,
+            author: post.data.author,
+            permalink: post.data.permalink
+          }
+        })
+        resolve(data);
+      })
+      .catch(function() {
+        reject('Could not fetch data.');
+      })
+    } else {
+      reject('url should not be undefined, empty or null.');
+    }
+  });
+}
+
+App.prototype.renderView = function (str, data) {
+  var fn = !/\W/.test(str) ?
+    this.tmplCache[str] = this.tmplCache[str] ||
+      this.renderView(document.getElementById(str).innerHTML) :
+    new Function("obj",
+      "var p=[],print=function(){p.push.apply(p,arguments);};" +
+      "with(obj){p.push('" +
+      str
+        .replace(/[\r\t\n]/g, " ")
+        .split("<%").join("\t")
+        .replace(/((^|%>)[^\t]*)'/g, "$1\r")
+        .replace(/\t=(.*?)%>/g, "',$1,'")
+        .split("\t").join("');")
+        .split("%>").join("p.push('")
+        .split("\r").join("\\'")
+    + "');}return p.join('');");
+
+  return data ? fn( data ) : fn;
+};
+
+App.prototype.switchThemes = function (newTheme, onLoad) {
+  this.appEl.className = '';
+  this.appEl.classList.add(newTheme);
+
+  if (onLoad) return
+
+  this.state.theme = newTheme;
+
+  this.saveState(Object.assign({}, this.state, {
+    theme: newTheme
+  }));
+};
+
+App.prototype.loadState = function () {
+  return new Promise(function(resolve, reject) {
+    return chrome.storage.local.get('state', function(result) {
+      if (result.state) {
+        var unserliazedState = JSON.parse(result.state)
+        if (unserliazedState !== undefined) {
+          resolve(unserliazedState)
+        }
+
+      } else {
+        resolve(undefined)
+      }
+    });
+  });
+};
+
+App.prototype.saveState = function (state) {
+  try {
+    const serializedState = JSON.stringify(state);
```

Comment:
Use `var` instead of `const`. I want to eventually move the entire project to es6, but for now let's stay with es5 for consistency.